webui/src/app/app.component.spec.ts

```diff
@@ -79,7 +84,43 @@ describe('AppComponent', () => {
     fixture.detectChanges();
   });
 
+  afterEach(() => {
+    testUser = null // need to reset to null every time
+  })
+
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should display login when user is null', () => {
+    expect(component.avatarImageUrl).toBeNull();
+    expect(fixture.debugElement.query(By.css('img.avatar'))).toBeNull()
+    expect(fixture.debugElement.query(By.css('mat-icon.avatar'))).toBeNull()
+    expect(fixture.debugElement.query(By.css('ngx-auth-firebaseui-providers'))).toBeTruthy()
+  });
+
+  it('should set avatar image url', () => {
+    testUser = {}
+    const expectedUrl = 'http://example.com';
+    const userSpy = {photoURL: expectedUrl};
+
+    expect(onAuthStateChangedSpy).toHaveBeenCalledWith(jasmine.any(Function));
+    const [ cb ] = onAuthStateChangedSpy.calls.mostRecent().args;
+    cb(userSpy);
+
+    expect(component.avatarImageUrl).toBe(expectedUrl);
+    expect(fixture.debugElement.query(By.css('img.avatar'))).toBeTruthy()
+    expect(fixture.debugElement.query(By.css('mat-icon.avatar'))).toBeNull()
+  });
+
+  it('should not set avatar image url when user photo url is null', () => {
+    testUser = {}
+    expect(onAuthStateChangedSpy).toHaveBeenCalledWith(jasmine.any(Function));
+    const [ cb ] = onAuthStateChangedSpy.calls.mostRecent().args;
+    cb({photoURL: null});
+
+    expect(component.avatarImageUrl).toBeNull();
+    expect(fixture.debugElement.query(By.css('img.avatar'))).toBeNull()
+    expect(fixture.debugElement.query(By.css('mat-icon.avatar'))).toBeTruthy()
```

Comment:
This is failing and I can't figure out why. @zsxking I could use your help debugging the failure.